igvc_navigation/src/mapper/mapper.cpp

```diff
@@ -0,0 +1,197 @@
+
+  igvc::getParam(pNh, "probability_model/camera/prob_miss", camera_probability_model_.prob_miss);
+  igvc::getParam(pNh, "probability_model/camera/prob_hit", camera_probability_model_.prob_hit);
+
+  igvc::getParam(pNh, "filters/ground_filter/enable", use_ground_filter_);
+  igvc::getParam(pNh, "filters/ground_filter/ransac/iterations", ground_filter_options_.ransac_options.iterations);
+  igvc::getParam(pNh, "filters/ground_filter/ransac/iterations", ground_filter_options_.ransac_options.iterations);
+  igvc::getParam(pNh, "filters/ground_filter/ransac/distance_threshold",
+                 ground_filter_options_.ransac_options.distance_threshold);
+  igvc::getParam(pNh, "filters/ground_filter/ransac/eps_angle", ground_filter_options_.ransac_options.eps_angle);
+  igvc::getParam(pNh, "filters/ground_filter/fallback/plane_distance",
+                 ground_filter_options_.fallback_options.plane_distance);
+
+  igvc::getParam(pNh, "filters/empty/start_angle", empty_filter_options_.start_angle);
+  igvc::getParam(pNh, "filters/empty/end_angle", empty_filter_options_.end_angle);
+  igvc::getParam(pNh, "filters/empty/miss_cast_distance", empty_filter_options_.miss_cast_distance);
+
+  igvc::getParam(pNh, "filters/empty_image/blur/kernel", process_image_options_.blur.kernel);
+  igvc::getParam(pNh, "filters/empty_image/blur/sigma", process_image_options_.blur.sigma);
+  igvc::getParam(pNh, "filters/empty_image/threshold", process_image_options_.threshold.threshold);
+
+  igvc::getParam(pNh, "filters/behind/filter_angle", behind_filter_options_.angle);
+  igvc::getParam(pNh, "filters/behind/distance", behind_filter_options_.distance);
+
+  igvc::getParam(pNh, "filters/combined_map/blur/kernel", combined_map_options_.blur.kernel);
+
+  igvc::getParam(pNh, "octree/resolution", resolution_);
+
+  igvc::getParam(pNh, "node/use_lines", use_lines_);
+
+  invertMissProbabilities();
+
+  octomapper_ = std::make_unique<Octomapper>(pNh);
+  octomapper_->create_octree(pc_map_pair_);
+  octomapper_->create_octree(camera_map_pair_);
+
+  camera_line_pub_ = pNh.advertise<pcl::PointCloud<pcl::PointXYZ>>("/mapper/camera_lines", 1);
+  camera_projection_pub_ = pNh.advertise<pcl::PointCloud<pcl::PointXYZ>>("/mapper/camera_projection", 1);
+  filtered_pc_pub_ = pNh.advertise<pcl::PointCloud<pcl::PointXYZ>>("/mapper/filtered_pc", 1);
+  empty_pc_pub_ = pNh.advertise<pcl::PointCloud<pcl::PointXYZ>>("/mapper/empty_pc", 1);
+}
+
+void Mapper::insertLidarScan(const pcl::PointCloud<pcl::PointXYZ>::ConstPtr& pc, const tf::Transform& odom_to_lidar)
+{
+  pcl::PointCloud<pcl::PointXYZ> empty_pc{};
+  pcl::PointCloud<pcl::PointXYZ> filtered_pc{};
+
+  MapUtils::getEmptyPoints(*pc, empty_pc, angular_resolution_, empty_filter_options_);
+  MapUtils::filterPointsBehind(*pc, filtered_pc, behind_filter_options_);
+
+  pcl_ros::transformPointCloud(filtered_pc, filtered_pc, odom_to_lidar);
```

Comment:
Shouldn't this transform be lidar_to_odom since the pointcloud is in the /lidar frame?

Reply:
fixed.